### PR TITLE
Lf 2842 reformat date display based on regional preference

### PR DESCRIPTION
**Pull request by gursimran-singh**
**Description**

Reformat date to display in correct format as per user's browser localization.

Jira link: https://lite-farm.atlassian.net/browse/LF-2842

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

**Steps to recreate:**

1.  Log in to your farm
2. Navigate to finances
3. Filter for a date range with expenses
4. Click “Other expenses”

**Desired:**

- Date is displayed in correct format for user browser localization

**Observed:**
- Displays in yyyy-mm-dd

Follow [this](https://www.aleksandrhovhannisyan.com/blog/changing-locale-in-chrome-with-dev-tools/) guide to change locale in chrome for testing.

- [x] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added the GNU General Public License to all new files

### CODE REVIEW

File: packages/webapp/src/containers/Finances/OtherExpense/index.jsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ class OtherExpense extends Component {
         let amount = parseFloat(e.value);
         subTotal += amount;
         detailedHistory.push({
-          date: moment(e.expense_date),
+          date: moment(e.expense_date).toDate().toLocaleDateString(),
```
Comment (SayakaOno) [inline packages/webapp/src/containers/Finances/OtherExpense/index.jsx:132]: I think it is better to do this on line 181. (That is what `Cell` is for)
`Cell: (d) => <span>{d.value.toDate().toLocaleDateString()}</span>`

Is this `mm/dd/yyyy` the correct format in North America?

Comment (gursimran-singh) [reply packages/webapp/src/containers/Finances/OtherExpense/index.jsx]: North America uses `mm/dd/yyyy` date format. We can check it here https://iso.mit.edu/americanisms/date-format-in-the-united-states/

Comment (SayakaOno) [reply packages/webapp/src/containers/Finances/OtherExpense/index.jsx]: 👍 I will never get used to this format 😵‍💫

**Review by SayakaOno — APPROVED** [review]
Looks good to me!

**Review by Duncan-Brain — APPROVED** [review]
Looks good to me!

Thanks for sharing the testing resource for locales: https://www.aleksandrhovhannisyan.com/blog/changing-locale-in-chrome-with-dev-tools/

**Pull request merged**